secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/mtls/CertificateResolver.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.mtls;
+
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+import org.forgerock.http.protocol.Request;
+import org.forgerock.services.context.Context;
+
+/**
+ * Resolves of a client's mTLS certificate
+ */
+public interface CertificateResolver {
+
```

Comment:
Resolve indicates a decision is being made. CertificateRetriever or CertificateFetcher might be better?